config/options.go

```diff
@@ -414,6 +414,10 @@ func (o *Options) Validate() error {
 		return errors.New("config: shared-key cannot be empty")
 	}
 
+	if o.SharedKey != strings.TrimSpace(o.SharedKey) {
```

Comment:
Discussion to be had as to whether we just trim it instead of dying.  I feel like there will be less mystery failures if we fail hard if the format is off.

Reply:
I agree. I think it makes sense to fail instead of silently change the value of the secret.